### PR TITLE
DM-33783: Remove internal multiprocessing from DefineVisitsTask.

### PR DESCRIPTION
**Pull request by TallJimbo**
This had stopped working a while ago, if it ever worked at all, and
with us now creating raw WCSs from Registry's DimensionRecords instead
of reading raw files, it's not worth trying to get working.

## Checklist

- [x] ran Jenkins
- [x] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

File: python/lsst/obs/base/defineVisits.py [hunk 1]
```diff
@@ -525,13 +517,9 @@ def run(
...
             differs from the new one.
         """
-        # Set up multiprocessing, if desired.
-        if pool is None and processes > 1:
-            pool = Pool(processes)
-        mapFunc = map if pool is None else pool.imap_unordered
         # Normalize, expand, and deduplicate data IDs.
         self.log.info("Preprocessing data IDs.")
-        dataIds = set(mapFunc(self._expandExposureId, dataIds))
+        dataIds = set(map(self._expandExposureId, dataIds))
```
Comment (timj) [inline python/lsst/obs/base/defineVisits.py:522]: Can we do this as the more obvious:
```python
dataIds = {self._expandExposureId(d) for d in dataIds}
```
?

File: python/lsst/obs/base/defineVisits.py [hunk 2]
```diff
@@ -573,7 +561,7 @@ def run(
         # This is the only parallel step, but it _should_ be the most expensive
         # one (unless DB operations are slow).
         self.log.info("Computing regions and other metadata for %d visit(s).", len(definitions))
-        allRecords = mapFunc(self._buildVisitRecordsSingle, zip(definitions, itertools.repeat(collections)))
+        allRecords = map(self._buildVisitRecordsSingle, zip(definitions, itertools.repeat(collections)))
```
Comment (timj) [inline python/lsst/obs/base/defineVisits.py:564]:
```python
allRecords = [self._buildVisitRecordsSingle(def, collections) for def in definitions]
```
? Seems like we no longer need to be using `itertools.repeat()` there.

**Review by timj — APPROVED** [review]
Minor comments on the `map` usage.

File: python/lsst/obs/base/defineVisits.py (hunk 2 above)
Comment (TallJimbo) [reply python/lsst/obs/base/defineVisits.py]: This also let me remove `_buildVisitRecordsSingle`, which only existed to unpack the `tuple` returned by `zip` into a pair of arguments to `_buildVisitRecords`.

**Pull request merged**